04_financial-security/city-debt-coll-shares.qmd

```diff
@@ -0,0 +1,197 @@
+---
+title: "Debt in Collections City-Level Shares"
+author: "Manuel Alcalá Kovalski"
+date: today
+format:
+  html:
+    toc: true
+    toc_float: true
+    embed-resources: true
+    code-fold: show
+execute: 
+  warning: false
+editor_options: 
+  chunk_output_type: console
+---
+
+## Overview
+
+This document processes city-level debt in collections data from the 
+Urban Institute's [Financial Health & Wealth dashboard](https://apps.urban.org/features/financial-health-wealth-dashboard/?location=4803000&year=2021&metric=delinquent_debt&subgroup=Residents+of+color)  for 2019, 2021, 2022, and 2023. 
+
+The processed data is used for the Upward Mobility Metrics Dashboard.
+
+```{r setup}
+library(tidyverse)
+library(readxl)
+library(here)
+library(tigris)
+# Set output path
+output_path <- here("04_financial-security/final")
+```
+
+Get state abbreviations & FIPS codes using `tigris` and place FIPS codes from IPUMS
+```{r}
+state_fips <- fips_codes |>
+    distinct(state, state_code, state_name) |>
+    rename(stusab = state, state = state_code)
+
+
+temp_file_ipums <- tempfile(fileext = ".xlsx")
+download.file(
+    url = "https://usa.ipums.org/usa/resources/volii/large_place_PUMA2010_match_summary.xlsx",
+    destfile = temp_file_ipums,
+    mode = "wb"
+)
+
+
+place_fips <- read_excel(temp_file_ipums) |>
+    rename(
+        place_fips = `Place FIPS Code`,
+        city = `IPUMS CITY Label`
+    ) |>
+    # Fix specific city labels to match dashboard data
+    mutate(
+        city = case_when(
+            `State Name` == "Arizona" & `Census Place Name` == "Gilbert town" ~ "Gilbert, AZ",
+            `State Name` == "Hawaii" & `Census Place Name` == "Urban Honolulu CDP" ~ "Honolulu, HI",
+            `State Name` == "Kansas" & `Census Place Name` == "Overland Park city" ~ "Overland Park, KS",
+            `State Name` == "Nevada" & `Census Place Name` == "Paradise CDP" ~ "Paradise, NV",
+            `State Name` == "Arizona" & `Census Place Name` == "Scottsdale city" ~ "Scottsdale, AZ",
+            `State Name` == "Nevada" & `Census Place Name` == "Sunrise Manor CDP" ~ "Sunrise Manor, NV",
+            TRUE ~ city
+        )
+    ) |>
+    filter(city != "[Not identified in any sample]") |>
+    select(place_fips, city)
+
+```
+
+## Import and process Urban Data Catalog file 
+
+Download and read Excel file from Financial Health & Wealth Dashboard
+
+```{r}
+overall_debt_city <-
+  read_csv("https://urban-data-catalog.s3.amazonaws.com/drupal-root-live/related-documents/city_2019_23.csv") |> 
+  filter(!is.na(state)) |> 
+  select(year,
+         city = city_name,
+         stusab = state,
+         subgroup,
+         share_debt_coll = delinquent_debt) |> 
+  filter(subgroup %in% c("Overall", "Residents of color", "White")) |> 
+     mutate(
+        subgroup = case_match(
+            subgroup,
+            "Overall" ~ "All",
+            "Residents of color" ~ "Majority Non-White",
+            "White" ~ "Majority White"
```

Comment:
We had to write bandaid code to fix this last update. https://github.com/UI-Research/mobility-from-poverty/blob/main/10_construct-database/63_construct_place_race-share-geography.qmd

Can we update this to make sure it aligns with the labels here: https://github.com/UI-Research/mobility-from-poverty/blob/main/data/63_mobility-metrics_place_race-share_longitudinal.csv